plugins/confluent/credentials.go

```diff
@@ -0,0 +1,94 @@
+package confluent
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+const (
+	MDS_URL                      = sdk.FieldName("MDS URL")
+	ConfluentCloudCredentials    = sdk.CredentialName("Confluent Cloud Credentials")
+	ConfluentPlatformCredentials = sdk.CredentialName("Confluent Platform Credentials")
+)
+
+func CloudCredentials() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          ConfluentCloudCredentials,
+		DocsURL:       sdk.URL("https://docs.confluent.io/confluent-cli/current/command-reference/confluent_login.html"),
+		ManagementURL: sdk.URL("https://confluent.cloud"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Username,
+				MarkdownDescription: "Email used to authenticate to Confluent Cloud.",
+				Secret:              false,
+				Optional:            false,
+			},
+			{
+				Name:                fieldname.Password,
+				MarkdownDescription: "Password used to authenticate to Confluent Cloud.",
+				Secret:              true,
+				Optional:            false,
+			},
+			{
+				Name:                fieldname.Organization,
+				MarkdownDescription: "Organization to use with Confluent Cloud.",
+				Secret:              false,
+				Optional:            true,
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultCloudEnvVarMapping),
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultCloudEnvVarMapping),
+		)}
+}
+
+func PlatformCredentials() schema.CredentialType {
+	return schema.CredentialType{
+		Name:    ConfluentPlatformCredentials,
+		DocsURL: sdk.URL("https://docs.confluent.io/confluent-cli/current/command-reference/confluent_login.html"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Username,
+				MarkdownDescription: "Username used to authenticate to Confluent Platform.",
+				Secret:              false,
+				Optional:            false,
+			},
+			{
+				Name:                fieldname.Password,
+				MarkdownDescription: "Password used to authenticate to Confluent Platform.",
+				Secret:              true,
+				Optional:            false,
+			},
+			{
+				Name:                MDS_URL,
+				MarkdownDescription: "MDS used to authenticate to Confluent Platform.",
+				Secret:              false,
+				Optional:            true,
+			},
+			{
+				Name:                fieldname.Certificate,
+				MarkdownDescription: "TLS Certificate File used to authenticate to Confluent Platform MDS.",
+				Secret:              false,
+				Optional:            true,
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultPlatformEnvVarMapping),
```

Comment:
I would propose, for better code readability, that we comment out all the bits that relate to Confluent Platform, with a TODO about revisiting this once the Shell Plugins ecosystem adds support for multiple credential types per plugin.

Reply:
This is all commented out now and a TODO left about it.